src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -945,7 +945,7 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
             const auto& input = dep.first;
             auto input_layout = input->get_output_layout();
             // Change input data type of concat node from input format to output format
-            if (input_layout.format != output_layout.format) {
+            if (concat_node.get_preferred_impl_type() != cldnn::impl_types::onednn && output_layout.format != input_layout.format) {
```

Comment:
* Why did it cause functional issue?
* Could you add unittest for such behavior?